mongoengine/queryset/base.py

```diff
@@ -722,7 +722,7 @@ def with_id(self, object_id):
         :param object_id: the value for the id of the document to look up
         """
         queryset = self.clone()
-        if not queryset._query_obj.empty:
+        if bool(queryset._query_obj):
```

Comment:
`not (not bool(queryset._query_obj.query))` -> `bool(queryset._query_obj)`

Reply:
I believe we can simply use 
`if queryset._query_obj:`

This should make use of __bool__ naturally

Reply:
Sounds good, I was just applying what the deprecation  warning says: `"'empty' property is deprecated in favour of using 'not bool(filter)'"`, but we don't have to cast it explicitly like you said.

Reply:
indeed, the warning was misleading :D